cinema/tests/test_movie_veiw_set_api.py

```diff
@@ -0,0 +1,222 @@
+        genre_one = sample_genre(name="Sci-Fi")
+        genre_two = sample_genre(name="Comedy")
+
+        movie = sample_movie()
+        movie_one = sample_movie(
+            title="first", description="second test description", duration=70
+        )
+        movie_two = sample_movie(
+            title="test", description="test description", duration=60
+        )
+
+        movie_one.genres.add(genre_one)
+        movie_two.genres.add(genre_two)
+
+        response_filter_title = self.client.get(
+            MOVIE_URL, {"genres": f"{genre_one.id},{genre_two.id}"}
+        )
+
+        serializer = MovieListSerializer(movie)
+        serializer_with_genre_one = MovieListSerializer(movie_one)
+        serializer_with_genre_two = MovieListSerializer(movie_two)
+
+        self.assertIn(serializer_with_genre_one.data, response_filter_title.data)
+        self.assertIn(serializer_with_genre_two.data, response_filter_title.data)
+        self.assertNotIn(serializer.data, response_filter_title.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(Genre.objects.create(name="Sci-Fi"))
+
+        url = reverse("cinema:movie-detail", args=(movie.id,))
+
+        response = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "Sample movie",
+            "description": "Sample description",
+            "duration": 90,
+        }
+
+        response = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="user@example.com",
+            password="testpassword",
+            is_staff=True,
+        )
+
+        self.client.force_authenticate(self.user)
+
+    def test_create_movie(self):
+        payload = {
+            "title": "Sample movie",
+            "description": "Sample description",
+            "duration": 90,
+        }
+
+        response = self.client.post(MOVIE_URL, payload)
+        movie = Movie.objects.get(id=response.data["id"])
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        for key in payload:
+            self.assertEqual(payload[key], getattr(movie, key))
+
+    def test_create_movie_with_genres_actors(self):
+        sample_actor()
+        sample_genre()
+
+        actor = Actor.objects.get(id=1)
+        genre = Genre.objects.get(id=1)
+
+        payload = {
+            "title": "Sample movie",
+            "description": "Sample description",
+            "duration": 90,
+            "genres": actor.id,
+            "actors": genre.id,
```

Comment:
In the `test_create_movie_with_genres_actors` method, the payload incorrectly assigns `actor.id` to `genres` and `genre.id` to `actors`. This should be corrected to `"genres": genre.id` and `"actors": actor.id` to match the intended functionality.